Update givaro submodule w/ fix for GCC 15 bug

d-torrance opened the pull request:
The givaro developers have merged https://github.com/linbox-team/givaro/pull/233, which fixes the "no member named 'Low' in 'rint<K>'" compilation failure we were getting with GCC 15.

Closes: #3507

d-torrance commented:
The givaro devs just tagged a new release candidate (4.2.1rc0), so I've updated this PR to use that commit instead.

mikestillman reviewed (approved):
This one looks ready to go.  @mahrud, do you think so too?

@d-torrance Do we need to change anything else?  Is this working on both cmake and autotools?

d-torrance commented:
Yes, it should be good.  In particular, the build now works when we build givaro ourselves with GCC 15.

d-torrance commented:
Givaro just released v4.2.1, so I've updated the submodule commit again to reflect that.

I added a temporary commit that should build givaro (and ffpack) and use it in the GitHub builds to double check that everything's still working.

d-torrance commented:
The builds all passed (with the exception of the autotools Ubuntu builds, but that's because of an autotools bug I should fix in another PR), so I'm going to remove the last commit.
